File: machsuite/machsuite-fft-strided/parse.py
import json

# Rather ugly parsing code

var_names = ["rel", "img", "rel_twid", "img_twid"]
in_data = {}
out_data = {}

with open("input.data", "r+") as f:
    idx = 0
    arr = []
    for line_raw in f:
        line = line_raw.strip()
        if line != "%%":
            print(line)
            arr.append(float(line))
        else:
            if idx != 0:
                in_data[var_names[idx-1]] = {
                "data": arr,
                "format": {
                    "numeric_type": "bitnum",
                    "is_signed": False,
                    "width": 32
                    }
                }
                arr = []
            idx += 1
    in_data["img_twid"] = {
                "data": arr,
                "format": {
                    "numeric_type": "bitnum",
                    "is_signed": False,
                    "width": 32
                    }
                }


with open("check.data", "r+") as f:    
    idx = 0
    arr = []
    for line_raw in f:
        line = line_raw.strip()
        if line != "%%":
            arr.append(float(line))
        else:
            if idx != 0:
                out_data[var_names[idx-1]] = {
                "data": arr,
                "format": {
                    "numeric_type": "bitnum",
                    "is_signed": False,
                    "width": 32
                    }
                }
                arr = []
            idx += 1
    out_data["img"] = {
                "data": arr,
                "format": {
                    "numeric_type": "bitnum",
                    "is_signed": False,
                    "width": 32
                    }
                }

out_data["rel_twid"] = in_data["rel_twid"]
out_data["img_twid"] = in_data["img_twid"]


############ Write float test data

with open("fft.fuse.double.json", "w+") as f:
    f.write(json.dumps(in_data))

with open("fft.expect.double.json", "w+") as f:
    f.write(json.dumps(out_data))

File: machsuite/machsuite-spmv-crs/parse.js
const fs = require("fs")

const input = fs.readFileSync('./input.data', 'utf8').split("%%").slice(1).map(a => a.split("\n")).map(a => {
  const data = a.filter(a => a)
  return data.map(a => parseFloat(a))
})

const check_data = fs.readFileSync("./check.data", 'utf8').split("\n").slice(1, 495).map(a => parseFloat(a))

const format = {
    "numeric_type": "bitnum",
    "is_signed": false,
    "width": 32
  }
  
const val = { "data": input[0], format };
const cols = { "data": input[1], format };
const row_delimiters = { "data": input[2], format };
const vec = { "data": input[3], format };
const out = { "data": Array.from(check_data).fill(0), format };

const outdata = { val, cols, row_delimiters, vec, out };
fs.writeFileSync('./spmv.in.double.json', JSON.stringify(outdata), 'utf8');
fs.writeFileSync('./spmv.expect.double.json', JSON.stringify({ "out": {"data": check_data, format} }), 'utf8');
